### ISSUE TITLE
Option to dump metadata in a machine-readable format

### ISSUE THREAD

**Issue by CyberTailor** [haskell/cabal#9056]
**Feature request:**
Add a new command to dump metadata as JSON or any other machine-readable format.

Output could be used by tools like [upstream-ontologist](https://github.com/jelmer/upstream-ontologist). There are 3rd-party `.cabal` parsers but they're all wrong - the format is hard to parse.

**Comment by ulysses4ever** [comment]
Welcome to the Cabal bug tracker and thanks for your interest! Could you make your request a bit more precise: what kind of metadata are you interested in?

Have you looked into plan.json? It's a file generated by cabal on almost any action in a clean project in the dist-newstyle directory (somewhere deep down in it). It may be just enough for many kinds of cases.

**Comment by CyberTailor** [comment]
> what kind of metadata are you interested in?

Description, license, authors, URLs.

> Have you looked into plan.json?

Yes. It contains information about dependencies, not project metadata.

**Comment by ulysses4ever** [comment]
So, it looks like you want a parser of `.cabal` files that could also dump JSON. Dumping JSON should be easy enough, the hard part is the parser. What parsers have you tried? Did you try the Cabal's own parser? Cabal is a library and it should expose everything necessary.

**Comment by andreabedini** [comment]
A short term solution can be put together with a bunch of `ToJSON` orphan instances on top of the types exposed by `Cabal-syntax`. That's what I did in https://github.com/andreabedini/Cabal-syntax-json for example. I made that to figure out the `FieldGrammar` machinery, but I am still not sure how to use it :P

**Comment by CyberTailor** [comment]
> What parsers have you tried?

[go-cabal-parser](https://github.com/goncharovnikita/go-cabal-parser) and upstream-ontologists's
